baseplate/integration/pyramid/csrf.py

```diff
@@ -0,0 +1,129 @@
+            CSRF token from on a request.  The default is 'csrf_token'.
+        :param int version: (Optional) Version of the CSRF token.  The default
+            is 1.
+        :param datetime.timedelta max_age: (Optional) The maximum age that the
+            signature portion of the CSRF token is valid.  The default value is
+            one hour.
+        """
+
+        VERSION = 1
+
+        def __init__(self, secrets, secret_path,
+                     account_id_from_request_fn=_account_id_from_edge_context,
+                     param='csrf_token', max_age=timedelta(hours=1)):
+            self._secrets = secrets
+            self._secret_path = secret_path
+            self._account_id_from_request = account_id_from_request_fn
+            self._param = param
+            self._max_age = max_age
+
+        def _get_secret(self):
+            return self._secrets.get_versioned(self._secret_path)
+
+        def new_csrf_token(self, request):
+            """Return a new CSRF token.
+
+            You will need to call `pyramid.csrf.new_csrf_token` to get a new
+            CSRF token when rendering forms, you should not re-use CSRF tokens
+            when using this StoragePolicy since these tokens expire and are
+            difficult to selectively invalidate.
+            """
+            prefix, payload = _make_csrf_token_payload(
+                version=self.VERSION,
+                account_id=self._account_id_from_request(request),
+            )
+            signature = make_signature(self._get_secret(), payload, self._max_age)
+            return '.'.join([prefix, signature.decode('utf-8')])
+
+        def get_csrf_token(self, request):
+            """Return the currently active CSRF token from the request params.
+
+            This will not generate a new one if none is supplied like some of
+            the default ones in pyramid do.
+
+            This is called automatically by Pyramid if you have configured it
+            to require CSRF.
+            """
+            return request.params.get(self._param)
+
+        def check_csrf_token(self, request, supplied_token):
+            """Return True if the supplied_token is valid.
+
+            This is called automatically by Pyramid if you have configured it
+            to require CSRF.
+            """
+            try:
+                token_version, sep, signature = supplied_token.partition('.')
+            except Exception:
+                return False
+
+            if sep != ".":
+                return False
+
+            _, payload = _make_csrf_token_payload(
+                version=token_version,
+                account_id=self._account_id_from_request(request),
+            )
+
+            try:
+                validate_signature(self._get_secret(), payload, signature)
+            except SignatureError:
+                return False
+
+            return True
+except ImportError:
```

Comment:
it might be safer to only have the specific imports that might legit be missing in the try/except. this might otherwise mask weirder issues.